docs/architecture/psa-migration/md-cipher-dispatch.md

```diff
@@ -477,3 +499,54 @@ The architecture can be extended to support `MBEDTLS_PSA_CRYPTO_CLIENT` with a l
 
 * Compile-time dependencies: instead of checking `defined(MBEDTLS_PSA_CRYPTO_C)`, check `defined(MBEDTLS_PSA_CRYPTO_C) || defined(MBEDTLS_PSA_CRYPTO_CLIENT)`.
 * Implementers of `MBEDTLS_PSA_CRYPTO_CLIENT` will need to provide `psa_can_do_hash()` (or a more general function `psa_can_do`) alongside `psa_crypto_init()`. Note that at this point, it will become a public interface, hence we won't be able to change it at a whim.
+
+### Cipher light
+
+#### Definition
+
+**Note:** this definition is tentative an may be refined when implementing and
+testing, based and what's needed by internal users of Cipher light.
+
+Cipher light will be automatically enabled in `build_info.h` by modules that
+need it. (Tentative list: PEM, PCKS12, PKCS5, CTR\_DRBG, CCM, CMAC, GCM,
+NIS\_KW, PSA Crypto.) Note: some of these modules currently depend on the
+full `CIPHER_C` (enforced by `check_config.h`); this hard dependency would be
+replace by the above auto-enablement.
+
+Cipher light includes:
+- info functions;
+- support for block ciphers in ECB mode (to be confirmed: supporting one block
+  at a time could be enough);
+- support for block ciphers in CBC mode with no padding (to be confirmed: do
```

Comment:
PKCS7 padding and no-padding should be supported. Most users of CBC need PKCS7 padding; TLS is a notable exception (it uses no-padding mode).

Most modern systems don't need CBC at all. Do we really need CBC in cipher-light? Should we have a cipher-light with only ECB encryption and a cipher-mid with both directions of ECB and CBC?

Reply:
> PKCS7 padding and no-padding should be supported. Most users of CBC need PKCS7 padding; TLS is a notable exception (it uses no-padding mode).
> 
We don't need to support TLS here: with `USE_PSA_CRYPTO` it won't use Cipher, and if you want to use TLS in a driver-only build, you need to enable that.

> Most modern systems don't need CBC at all. Do we really need CBC in cipher-light? Should we have a cipher-light with only ECB encryption and a cipher-mid with both directions of ECB and CBC?

Having cipher-light that only supports what GCM and CCM need (that is, ECB encrypt only), and then cipher-medium that support the needs of other modules (that is, CBC), sounds like a good idea. I'll look into it further.

Reply:
> We don't need to support TLS here: with USE_PSA_CRYPTO it won't use Cipher,

TLS doesn't call cipher directly, but PSA crypto calls cipher for every cipher/AEAD mode except ChachaPoly.

Reply:
Why not have cipher light with only the basic transformation (it's not really ECB mode, since that can take a message of any size) in forward (encryption) mode if possible, but with decryption enabled if required? Then the ciphers wanted (whether CBC with or without padding, CTR, or something better like GCM) are built on top of that?

Reply:
CBC is part of cipher. We're designing subsets of the existing cipher interface, not a brand new interface (for that we already have PSA). A build with `MBEDTLS_CIPHER_LIGHT && !MBEDTLS_CIPHER_C` either includes CBC or it doesn't.

Reply:
The API could be present but return `UNSUPPORTED` when you request an padding more that isn't included.